packages/sdk-core/src/BacktraceCoreClient.ts

```diff
@@ -156,8 +165,14 @@ export abstract class BacktraceCoreClient {
      * Add attribute to Backtrace Client reports.
      * @param attributes key-value object with attributes.
      */
-    public addAttribute(attributes: Record<string, unknown>) {
-        this._attributeProvider.add(attributes);
+    public addAttribute(attributes: Record<string, unknown>): void;
+    public addAttribute(attributes: () => Record<string, unknown>): void;
+    public addAttribute(attributes: Record<string, unknown> | (() => Record<string, unknown>)) {
+        if (typeof attributes === 'function') {
```

Comment:
What do you think about moving this detail to attribute manager? I think this should be the internal logic that belongs to the manager.